operators/vtk_renderer/vtk_renderer.cpp

```diff
@@ -0,0 +1,189 @@
+  spec.param(this->in_annotations,
+             "in_annotations",
+             "Input Annotations",
+             "Input Annotations",
+             &in_annotations);
+  spec.param(this->in_videostream,
+             "in_videostream",
+             "Input Videostream",
+             "Input Videostream",
+             &in_videostream);
+
+  spec.param(
+      this->window_name, "window_name", "Window Name", "Window Name", std::string("VTK (Kitware)"));
+  spec.param(this->width, "width", "Window width", "Window width in px", 640u);
+  spec.param(this->height, "height", "Window height", "Window height in px", 480u);
+  spec.param(this->labels, "labels", "tools labels", "tools labels");
+}
+
+void VtkRendererOp::start() {
+  this->renderer_window->SetWindowName(this->window_name.get().c_str());
+  // Two layers, one for the background, one for the foreground
+  this->renderer_window->SetNumberOfLayers(2);
+  this->renderer_window->SetSize(this->width.get(), this->height.get());
+
+  // Foreground at the top-most layer.
+  this->foreground_renderer->SetLayer(1);
+  this->renderer_window->AddRenderer(this->foreground_renderer);
+
+  this->background_renderer->SetLayer(0);
+  this->background_renderer->InteractiveOff();
+  this->renderer_window->AddRenderer(this->background_renderer);
+  this->background_renderer->AddActor(this->image_actor);
+}
+
+void VtkRendererOp::compute(InputContext& op_input, OutputContext&, ExecutionContext& context) {
+  auto annotations = op_input.receive<gxf::Entity>("annotations").value();
+
+  // Render the annotations on the foreground layer.
+  auto scaled_coords_tensor = annotations.get<Tensor>("scaled_coords");
+  if (scaled_coords_tensor) {
+    std::vector<float> scaled_coords(scaled_coords_tensor->size());
+
+    // Copy the data from the tensor to the host.
+    CUDA_TRY(cudaMemcpy(scaled_coords.data(),
+                        scaled_coords_tensor->data(),
+                        scaled_coords_tensor->nbytes(),
+                        cudaMemcpyDeviceToHost));
+
+    this->foreground_renderer->RemoveAllViewProps();
+
+    // scale_coords comes in the format [X0 Y0 X1 Y1 ... Xn Yn]
+    // each numbered tuple represent a label (scissors, clipper...)
+    for (int i = 0; i < scaled_coords.size(); i += 2) {
+      if (scaled_coords[i] > 0) {
+        std::string label = labels.get()[i / 2];
+        float x = scaled_coords[i];
+        float y = scaled_coords[i + 1];
+        render_text_at_location(label, x, y, this->foreground_renderer);
+      }
+    }
+  }
+
+  // Render the videostream on the background layer.
+  auto videostream = op_input.receive<gxf::Entity>("videostream").value();
+  const auto videostream_tensor = videostream.get<Tensor>("");
+  if (videostream_tensor) {
+    holoscan::gxf::GXFTensor in_tensor_gxf{videostream_tensor->dl_ctx()};
+    auto& shape = in_tensor_gxf.shape();
+    const int x = shape.dimension(0);
+    const int y = shape.dimension(1);
+    const int z = shape.dimension(2);
+
```

Comment:
It might be nice to have some checks here that the image is really the expected format (uchar, component count, etc.) to avoid crashes when users try to input images with different formats.